includes/class-newspack-ads-model.php

```diff
@@ -508,5 +512,41 @@ function( $carry, $item ) {
 			[ 0, 0 ]
 		);
 	}
+
+	/**
+	 * Get ad targeting params for the current post or archive.
+	 *
+	 * @param array $ad_unit Ad unit to get targeting for.
+	 * @return array Associative array of targeting keyvals.
+	 */
+	public static function get_ad_targeting( $ad_unit ) {
+		$targeting = [];
+
+		if ( is_singular() ) {
+			// Add the post slug to targeting.
+			$slug = get_post_field( 'post_name' );
+			if ( $slug ) {
+				$targeting['slug'] = sanitize_text_field( $slug );
+			}
+
+			// Add the category slugs to targeting.
+			$categories = wp_get_post_categories( get_the_ID(), [ 'fields' => 'slugs' ] );
+			if ( ! empty( $categories ) ) {
+				$targeting['category'] = array_map( 'sanitize_text_field', $categories );
```

Comment:
If the post is uncategorized, the param will be `category=uncategorized`. Is this as it should be? Or should we exclude the param in this case?

Reply:
This is correct. "Uncategorized" is still a category:

<img width="588" alt="Screen Shot 2020-09-29 at 8 16 34 AM" src="https://user-images.githubusercontent.com/7317227/94578052-1b041100-022c-11eb-98d3-559d715b8263.png">

A site could rename that category or pick a different default category (e.g. "News"), so we'd want it to be available for targeting.